### PR TITLE
Integrating MongoDB with java

### LINKED ISSUE #4206: Integrating MongoDB with Java
#### Proposed title of article
How MongoDB integrate with JAVA 

#### Proposed article introduction
Here, we are going to learn how to connect MongoDB with java. MongoDB is based on a document model and can be written in C++, JavaScript and python.

#### Key takeaways 
After reading the article the reader should have learnt the following:
1. What MongoDB is
2. MongoDB vs Relational database structure
3. Required tools for connection
5. Creating MongoDB collection
6. Selecting a created MongoDB collection
7. What a document is and how it works with different anomalies(insert, update and delete)

#### Article quality
My topic involves basic knowledge that is easy to understand.

#### References
N/A

### PR DESCRIPTION
**Pull request by muthoni2-lucy**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #4206

### CODE REVIEW

File: Integrating MongoDB with Java/index.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
+**INTEGRATING MONGODB WITH JAVA**
+MongoDB is a no SQL database where records get stored in documents. It uses Jason syntax a JavaScript Object Notation to store data. When integrating MongoDB with java it uses a different syntax.
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:2]: `no SQL database`

It's NoSQL

Comment (srishilesh) [reply Integrating MongoDB with Java/index.md:2]: `Jason syntax`???

Comment (srishilesh) [reply Integrating MongoDB with Java/index.md:2]: `java` -> `Java`

File: Integrating MongoDB with Java/index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
+**INTEGRATING MONGODB WITH JAVA**
+MongoDB is a no SQL database where records get stored in documents. It uses Jason syntax a JavaScript Object Notation to store data. When integrating MongoDB with java it uses a different syntax.  
+In this article, you'll learn how to connect MongoDB with java using eclipse and learn crud (create, read, update and delete)  operations using Java and MongoDB Java Driver. MongoDB is based on a document model written in C++, JavaScript, and python.
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:3]: `crud` -> `CRUD`

File: Integrating MongoDB with Java/index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
+**INTEGRATING MONGODB WITH JAVA**
+MongoDB is a no SQL database where records get stored in documents. It uses Jason syntax a JavaScript Object Notation to store data. When integrating MongoDB with java it uses a different syntax.  
+In this article, you'll learn how to connect MongoDB with java using eclipse and learn crud (create, read, update and delete)  operations using Java and MongoDB Java Driver. MongoDB is based on a document model written in C++, JavaScript, and python.
+
+**Table of content**
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:5]: Use `###` markdown headers. Read the guidelines

File: Integrating MongoDB with Java/index.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+- [Update data in the collection](#Update-data-in-the-collection)
+
+- [Delete data in the collection](#Delete-data-in-the-collection)
+
+- [Delete the collection](#Delete-the-collection)
+
+-[Eclipse versus Intellij Idea in MongoDB integration with java](#Eclipse-versus-Intellij-Idea-in-MongoDB-integration-with-java)
+
+- [Conclusion](#Conclusion)
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:25]: Remove extra new lines

File: Integrating MongoDB with Java/index.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+MongoDB versus Relational Database Management System
+Connecting MongoDB with Java
+Creating MongoDB collection
+Documents in MongoDB
+Insert data
+Update data
+Delete data
+Drop collection
+Eclipse versus Intellij Idea in MongoDB integration with java
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:37]: Write full sentences. And, make them bulleted points.

File: Integrating MongoDB with Java/index.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+### Prerequisites
+To follow along with this tutorial you'll need the following:
+1. A basic understanding of the Java language
+2. Eclipse IDE installed In your OS
+3. MongoDB installed in your OS
+4. MongoDB java connector
+5. Java JDK
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:45]: Why do we have 2 prerequisite sections?

File: Integrating MongoDB with Java/index.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+### MongoDB versus Relational Database Management System
+1. MongoDB is a nonrelational database while Relational Database Management System is a relational database.
+2. MongoDB stores data using Jason docs, while RDBMS stores data using rows and columns.
+3. MongoDB uses collections while RDBMS uses tables.
+4. No schema definition is required in MongoDB, but a clear schema is required in RDBMS.
+5. Database in MongoDB is easy to scale, while in RDBMS options for scalability are more limited.
+6. RDBMS is slower than MongoDB.
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:53]: You have just stated facts. Explanation is needed

File: Integrating MongoDB with Java/index.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+       C:\Program Files\MongoDB\Server\5.0\bin>mongod --dbpath=mongodata
+4. After creating the folder, type mongod, and after running, it starts the MongoDB server. The current cmd prompt should not be closed.
+
+         C:\Program Files\MongoDB\Server\5.0\bin>mongod
+5. To run commands on MongoDB, you need to run the mongo command in the other cmd prompt type the command for MongoDB to open the shell.
+
+       C:\Program Files\MongoDB\Server\5.0\bin>mongo
+
+After running, you can show how many databases are currently available within your MongoDB
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:74]: Use ` `` ` for keywords, variables or single line code snippets.

File: Integrating MongoDB with Java/index.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+             import com.mongodb.DB;
+             import com.mongodb.DBCollection;
+             import com.mongodb.DBCursor;
+             import com.mongodb.DBObject;
+             import com.mongodb.MongoClient;
+             import com.mongodb.client.FindIterable;
+             import com.mongodb.client.MongoCollection;
+             import com.mongodb.client.MongoDatabase;
+             import com.mongodb.client.model.Filters;
+             import com.mongodb.client.model.Updates;
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:92]: Use markdown for code snippets

File: Integrating MongoDB with Java/index.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+	   collection.insertOne(doc2);
+	   System.out.println("Insert is completed");
+	   
+In the cmd shell prompt type 
+
+     myCursor=db.students.find({})
+    
+which will read the data that has been created as shown below.
+
+![cmd 3](image5.png)
```
Comment (srishilesh) [inline Integrating MongoDB with Java/index.md:130]: Image path is incorrect. Refer guidelines

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @muthoni2-lucy ,

Congratulations on submitting your first PR!

On a quick glance at your PR, I noticed that we might need several changes. I've suggested a few of them like:
- Your article must be inside `content/articles`.
- The article directory name must be lowercase separated by `-`, not `Integrating MongoDB with Java/index.md`
- The markdown formatting is not followed. Refer guidelines.
- `content/authors/Christine-irungu/Avatar.jpg` Every directory/file name must be in lowercase
- Perform a grammar check
- The code snippets are not enclosed within the markdown quotes
- The image paths are incorrect.

I would highly recommend you to take a step back, go through the contribution guidelines very carefully. You may also read merged PRs/published articles to see how other authors have formatted and structured the flow of the article.

Since your PR requires some major revisions, we will be closing this PR. However, you may work on the changes locally, create a new PR and ping me when you're done.

Visit the EngEd new contributor's docs [here](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md).

Here's a [link](https://github.com/section-engineering-education/engineering-education/pull/6210/commits/acc44893d4ec258f4e077b9ce8e3e2ca2b9d6a9f) of one of my articles which has little or no revisions to be made, to guide you on formatting.

Let me know if you have any questions.

Thanks.

**Comment by muthoni2-lucy** [comment]
Hey @srishilesh I pinged you at my new pr #6844

**Pull request closed without merging**